src/qibo/gates/channels.py

```diff
@@ -91,16 +91,17 @@ def to_choi(self, order: str = "row", backend=None):
                 self.coefficients += (p0,)
                 self.gates += (I(*self.target_qubits),)
 
-        super_op = np.zeros((4**self.nqubits, 4**self.nqubits), dtype="complex")
+        super_op = np.zeros((4**self.nqubits, 4**self.nqubits), dtype=complex)
+        super_op = backend.cast(super_op, dtype=super_op.dtype)
         for coeff, gate in zip(self.coefficients, self.gates):
             kraus_op = FusedGate(*range(self.nqubits))
             kraus_op.append(gate)
             kraus_op = kraus_op.asmatrix(backend)
-            kraus_op = vectorization(kraus_op, order=order)
+            kraus_op = vectorization(kraus_op, order=order, backend=backend)
             super_op += coeff * np.outer(kraus_op, np.conj(kraus_op))
             del kraus_op
 
-        super_op = backend.cast(super_op, dtype=super_op.dtype)
+        # super_op = backend.cast(super_op, dtype=super_op.dtype)
```

Comment:
Maybe these comments (and same in the following lines) are not needed.

Reply:
Thanks